test/functional/rpc_fundrawtransaction.py

```diff
@@ -907,5 +908,25 @@ def test_subtract_fee_with_presets(self):
         signedtx = self.nodes[0].signrawtransactionwithwallet(fundedtx['hex'])
         self.nodes[0].sendrawtransaction(signedtx['hex'])
 
+    def test_transaction_too_large(self):
+        self.log.info("Test fundrawtx where BnB solution would result in a too large transaction, but Knapsack would not")
+
+        self.nodes[0].createwallet("large")
+        wallet = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+        recipient = self.nodes[0].get_wallet_rpc("large")
+        outputs = {}
+        rawtx = recipient.createrawtransaction([], {wallet.getnewaddress(): 147.99899260})
+
+        # Make 1500 0.1 BTC outputs
+        # The amount that we target for funding is in the BnB range when these outputs are used.
+        # However if these outputs are selected, the transaction will end up being too large, so it shouldn't use BnB and instead fallback to Knapsack
+        for i in range(0, 1500):
+            outputs[recipient.getnewaddress()] = 0.1
+        wallet.sendmany("", outputs)
+        self.nodes[0].generate(10)
+
+        # Currently, the only solution is to select most of the coins just made, however this will result in a transaction that is too large
```

Comment:
I find it confusing the test claims it will have a successful knapsack run then doesn't?

Reply:
The fallback to knapsack behavior hasn't been implemented yet, so the only thing that should happen is we get an error. Originally this was part of a branch that did that fallback, but I removed it because I didn't like how it was implemented. I've updated the comment to reflect this.